WithSecure/withsecure-elements/ingest/parser.yml

```
@@ -0,0 +1,180 @@
+name: withsecure-elements
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: '{{original.message}}'
+        output_field: message
+  - name: read_event_date
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.clientTimestamp}}'
+        output_field: datetime
+  - name: set_common_fields
+  - name: set_malicious_file_fields
+    filter: '{{json_event.message.engine == "manualScanning" or json_event.message.engine == "fileScaning" or json_event.message.engine == "deepGuard"}}'
+  - name: deepguard_blocked_executable_file
+    filter: '{{json_event.message.engine == "deepGuard"}}'
+  - name: endpoint_detection_and_response
+    filter: '{{json_event.message.engine == "edr"}}'
+  - name: amsi
+    filter: '{{json_event.message.engine == "AMSI"}}'
+  - name: firewall_blocked_connection
+    filter: '{{json_event.message.engine == "firewall"}}'
+  - name: browsing_protection
+    filter: '{{json_event.message.engine == "browsingProtection" or json_event.message.engine == "reputationBasedBrowsing" or json_event.message.engine == "webContentControl"}}'
+  - name: connection_control
+    filter: '{{json_event.message.engine == "connectionControl"}}'
+stages:
+  set_common_fields:
+    actions:
+      - set:
+          '@timestamp': '{{read_event_date.datetime}}'
+          event.kind: 'event'
+          event.dataset: '{{json_event.message.engine}}'
+          event.action: '{{json_event.message.action}}'
+          event.reason: '{{json_event.message.details.reason}}'
+
+          agent.id: '{{json_event.message.device.id}}'
+          agent.name: '{{json_event.message.device.name}}'
+
+          host.ip: '{{json_event.message.details.hostIpAddress}}'
+          host.hostname: '{{json_event.message.device.name}}'
+
+          agent.type: 'WithSecure Agent'
+          organization.id: '{{json_event.message.organization.id}}'
+          organization.name: '{{json_event.message.organization.name}}'
+
+          process.executable: '{{json_event.message.details.creator}}'
+          process.hash.sha1: '{{json_event.message.details.creatorSha1}}'
+          process.command_line: '{{json_event.message.details.commandLine}}'
+          process.name: '{{json_event.message.details.process}}'
+
+          user.name: '{{json_event.message.details.userName}}'
+          url.full: '{{json_event.message.details.url}}'
+
+          file.hash.sha1: '{{json_event.message.details.sha1}}'
+          file.path: '{{json_event.message.details.path}}'
```

I think we can get the filename too:
```suggestion
          file.path: '{{json_event.message.details.path}}'
          file.name: '{{json_event.message.details.path | basename}}'
```